scripts/feature_tests/upgrade/upgrade_common.sh

```diff
@@ -144,12 +148,10 @@ EOF
 }
 
 function manage_node_taints() {
-    kubectl get secrets "${CLUSTER_NAME}"-kubeconfig -n "${NAMESPACE}" -o json | \
-      jq -r '.data.value'| base64 -d > /tmp/kubeconfig-"${CLUSTER_NAME}".yaml
```

Comment:
General comment: I would do the kubeconfig handling in upgrade_common.sh where it's needed. Will this work after pivoting? I don't have an answer yet.

Like this:

function call
: 
point_to_management_cluster
get_target_kubeconfig
point_to_target_cluster

 // run commands on target cluster
: 
// before exiting
point_to_management_cluster

Reply:
This way the actual test scripts would not need to worry about kubeconfig.

Reply:
Sorry, I missed this comment. 
If we take those rows to the common, then how does the test run ?
i.e The logic is in the non-common functions and order varies and cannot be pre-determined in the common functions file